python/composio/tools/env/filemanager/manager.py

```diff
@@ -213,8 +215,11 @@ def grep(
         else:
             pattern = Path(pattern)
 
+        exclude_paths = [self.working_dir / Path(ex) for ex in (exclude or [])]
```

Comment:
Consider adding error handling for cases where an excluded path doesn't exist or is not a directory. This would make the function more robust.